Tests/ArcGISToolkitTests/PreplannedMapModelTests.swift

```diff
@@ -218,4 +222,99 @@ class PreplannedMapModelTests: XCTestCase {
         XCTAssertFalse(model.canDownload)
         XCTAssertTrue(model.status.needsToBeLoaded)
     }
+    
+    @MainActor
+    func testDownloadingStatus() async throws {
+        class MockPreplannedMapArea: PreplannedMapAreaProtocol {
+            func retryLoad() async throws {}
+        }
+        
+        let mockArea = MockPreplannedMapArea()
+        let model = PreplannedMapModel(offlineMapTask: OfflineMapTask(onlineMap: Map()), mapArea: mockArea, portalItemID: "")!
+        await model.load()
+        
+        XCTAssertTrue(model.canDownload)
+        await model.downloadPreplannedMapArea()
+        model.updateDownloadStatus(for: model.result)
+        
+        guard case .downloading = model.status else {
+            XCTFail("PreplannedMapModel status is not \".downloading\".")
+            return
+        }
+        
+        XCTAssertFalse(model.canDownload)
+        XCTAssertFalse(model.status.needsToBeLoaded)
+    }
+    
+    @MainActor
+    func testDownloadFailureStatus() async throws {
+        class MockPreplannedMapArea: PreplannedMapAreaProtocol {
+            func retryLoad() async throws {}
+        }
+        
+        class MockPreplannedMapModel: PreplannedMapModel {
+            override var result: Result<MobileMapPackage, any Error>? {
+                get { _result }
+                set { _result = newValue }
+            }
+            var _result: Result<MobileMapPackage, any Error>?
+            
+            override func downloadPreplannedMapArea() async {
+                result = .failure(MappingError.invalidResponse(details: ""))
+            }
+        }
```

Comment:
by overriding the model's functionality, you are missing out on testing a lot of the model's functionality. Sometimes this might be ok. A better option is to use protocols and dependency injection when possible.